packages/api/tests/lib/FileUtils/test-parseSource.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const test = require('ava');
+const rewire = require('rewire');
+
+const fileUtils = rewire('../../../lib/FileUtils');
+const parseSource = fileUtils.__get__('parseSource');
+
+test('leaves file unchanged if it has no source', (t) => {
+  const file = {
+    key: 'key',
+    bucket: 'bucket',
+    path: 'path',
+    anythingelse: 'anything else'
+  };
+
+  const expected = { ...file };
+
+  const actual = parseSource(file);
+
+  t.deepEqual(expected, actual);
+});
+
+test('leaves file unchanged if key is not null', (t) => {
```

Comment:
The key is null here right? Same for line 38.

Reply:
In this case it's actually undefined, not null.  I could add another test to show one where, key has a value but bucket is null.    The getBucket and getKey functions return explicit null values, and so the file object has key: null and bucket: null, when the normal mode of operation doesn't find what it's looking for.  
I did it this way to show that leaving a value off the file, is not actually null.  I clarified by adding a test without a bucket or key and adding in not null values to these two,.

Reply:
7151f5e96